controllers/scan.js

```diff
@@ -21,19 +21,20 @@ async function post (req, res) {
     const newestBreach = new Date(b.BreachDate);
     return newestBreach-oldestBreach;
   });
-  
+
   if (req.body.featuredBreach) {
     featuredBreach = req.app.locals.breaches.find(breach => breach.Name.toLowerCase() === req.body.featuredBreach.toLowerCase()); 
 
     if (foundBreaches.find(breach => breach.Name === featuredBreach.Name)) {
       userAccountCompromised = true;
 
       if (foundBreaches.length > 1) {
-        foundBreaches.splice(foundBreaches.findIndex(breach => breach.Name === req.body.featuredBreach),1);
+        foundBreaches.splice(foundBreaches.indexOf(foundBreaches.find(breach => breach.Name === featuredBreach.Name),1));
+        foundBreaches.unshift(featuredBreach);
       }
 
-      if (foundBreaches.length === 1) {
-        foundBreaches = false;
+      if (foundBreaches.length === 1 && foundBreaches.find(breach => breach.Name === featuredBreach.Name)) {
```

Comment:
actually, this `foundBreaches.find(breach => breach.Name === featuredBreach.Name)` looks like a copy-pasta of L28 above (`if (foundBreaches.find(breach => breach.Name === featuredBreach.Name)) {...}`). We should probably just cache that result instead of doing the `.find()` again.